response.go

```diff
@@ -125,6 +126,46 @@ func ResponderFromResponse(resp *http.Response) Responder {
 	}
 }
 
+// ResponderFromMultipleResponses wraps an *http.Response list in a Responder.
+//
+// Each response will be returned in the order of the provided list.
+// If the responder is called more than the size of the provided list, an error
+// will be thrown.
+//
+// Be careful, except for responses generated by httpmock
+// (NewStringResponse and NewBytesResponse functions) for which there
+// is no problems, it is the caller responsibility to ensure the
+// response body can be read several times and concurrently if needed,
+// as it is shared among all Responder returned responses.
+//
+// For home-made responses, NewRespBodyFromString and
+// NewRespBodyFromBytes functions can be used to produce response
+// bodies that can be read several times and concurrently.
+func ResponderFromMultipleResponses(responses []*http.Response) Responder {
+	responseIndex := 0
+	mutex := sync.Mutex{}
+	return func(req *http.Request) (*http.Response, error) {
+		mutex.Lock()
+		defer mutex.Unlock()
+		defer func() { responseIndex = responseIndex + 1 }()
```

Comment:
```suggestion
		defer func() { responseIndex++ }()
```
Note that the mutex can be avoided using https://golang.org/pkg/sync/atomic/#AddInt64

Reply:
Considering the mutex part, I think it's mandatory using one since it avoid responding with twice the same array element before jumping of index.
ex without locking:
responses = [resp1, resp2]
Call1 & call2 are done in parallel
=> [call1] response = resp1
=> [call2] response = resp1
=> [call1] index++
=> [call2] index++

Reply:
You are right.